GDJS/tests/tests/spriteruntimeobject.js

```diff
@@ -70,18 +170,180 @@ describe('gdjs.SpriteRuntimeObject', function() {
+            name: '',
+            visibility: true,
+            effects: [],
+            cameras: [],
+
+            ambientLightColorR: 0,
+            ambientLightColorG: 0,
+            ambientLightColorB: 0,
+            isLightingLayer: false,
+            followBaseLayerCamera: true,
+          },
+        ],
+        variables: [],
+        r: 0,
+        v: 0,
+        b: 0,
+        mangledName: 'Scene1',
+        name: 'Scene1',
+        stopSoundsOnStartup: false,
+        title: '',
+        behaviorsSharedData: [],
+        objects: [],
+        instances: [],
+      });
+      const stepDurationInMilliseconds = 1000 / 60;
+      runtimeScene._timeManager.getElapsedTime = function () {
+        return stepDurationInMilliseconds;
+      };
+
+      const object = createObjectWithAnimationInScene(runtimeScene);
+      runtimeScene.addObject(object);
+
+      runtimeScene.renderAndStep(stepDurationInMilliseconds);
+
+      expect(object._frameElapsedTime).to.not.be(0);
+
+      object.setAnimation(1);
+
+      expect(object._frameElapsedTime).to.be(0);
+
+      runtimeScene.renderAndStep(stepDurationInMilliseconds);
+
+      expect(object._frameElapsedTime).to.not.be(0);
+    });
+
+    it('should reset the elapsed time on a frame when changing animation frame', () => {
+      const runtimeGame = new gdjs.RuntimeGame({
+        variables: [],
+        // @ts-expect-error ts-migrate(2740) FIXME: Type '{ windowWidth: number; windowHeight: number;... Remove this comment to see the full error message
+        properties: { windowWidth: 800, windowHeight: 600 },
+        resources: { resources: [] },
+      });
+      const runtimeScene = new gdjs.RuntimeScene(runtimeGame);
+      runtimeScene.loadFromScene({
+        layers: [
+          {
+            name: '',
+            visibility: true,
+            effects: [],
+            cameras: [],
+
+            ambientLightColorR: 0,
+            ambientLightColorG: 0,
+            ambientLightColorB: 0,
+            isLightingLayer: false,
+            followBaseLayerCamera: true,
+          },
+        ],
+        variables: [],
+        r: 0,
+        v: 0,
+        b: 0,
+        mangledName: 'Scene1',
+        name: 'Scene1',
+        stopSoundsOnStartup: false,
+        title: '',
+        behaviorsSharedData: [],
+        objects: [],
+        instances: [],
+      });
+      const stepDurationInMilliseconds = 1000 / 60;
+      runtimeScene._timeManager.getElapsedTime = function () {
+        return stepDurationInMilliseconds;
+      };
+
+      const object = createObjectWithAnimationInScene(runtimeScene);
+      runtimeScene.addObject(object);
+
+      runtimeScene.renderAndStep(stepDurationInMilliseconds);
+
+      expect(object._frameElapsedTime).to.not.be(0);
+
+      object.setAnimationFrame(2);
+
+      expect(object._frameElapsedTime).to.be(0);
+
+      runtimeScene.renderAndStep(stepDurationInMilliseconds);
+
+      expect(object._frameElapsedTime).to.not.be(0);
```

Comment:
In theory we should avoid reading an object internal state, but I guess this is fine.